scipy/linalg/tests/test_lapack.py

```diff
@@ -1654,3 +1654,138 @@ def test_getc2_gesc2():
         else:
             assert_array_almost_equal(desired_cplx.astype(dtype),
                                       x/scale, decimal=4)
+
+
+def generate_random_dtype_array(shape, dtype):
+    # generates a random matrix of desired data type of shape
+    if dtype in COMPLEX_DTYPES:
+        return (np.random.rand(*shape)
+                + np.random.rand(*shape)*1.0j).astype(dtype)
+    return np.random.rand(*shape).astype(dtype)
+
+
+@pytest.mark.parametrize("ddtype,dtype",
+                         zip(DTYPES, REAL_DTYPES + REAL_DTYPES))
+def test_pttrf_pttrs(ddtype, dtype):
+    np.random.seed(42)
+    # set test tolerance appropriate for dtype
+    rtol = 250*np.finfo(dtype).eps
+    atol = 100*np.finfo(dtype).eps
+    # n is the length diagonal of A
+    n = 10
+    # create diagonals according to size and dtype
+    if dtype in REAL_DTYPES:
+        # add 2 so that the matrix will be diagonally dominant
+        d = generate_random_dtype_array((n,), dtype) + 2
+    else:
+        # diagonal d should always be real.
+        # for complex add 4 so it will be dominant
+        d = (generate_random_dtype_array((n,), ddtype) + 4)
```

Comment:
Why is this one in parentheses?
Let's add +4 regardless of whether it's complex or not to eliminate the branching.